biodata-models/src/main/java/org/opencb/biodata/models/variant/Variant.java

```diff
@@ -219,7 +217,9 @@ public String getReference() {
 
     public void setReference(String reference) {
         this.reference = reference;
-        this.length = Math.max(reference.length(), alternate.length());
+        if (reference != null && alternate != null) {
+            this.length = Math.max(reference.length(), alternate.length());
```

Comment:
How often is getType called? Perhaps it would be safer to remove the type instance variable and calculate it in getType?